pkg/arvo/neo/cod/std/src/fil/diary.md

```diff
@@ -0,0 +1,297 @@
+
+## Diary frontend
+Like Counter, the Diary shrub just has two con files to convert to and from an HTMX frontend within the Sky browser.
+
+### /con/diary-htmx.hoon
+```hoon
+/@  diary  ::  name=@t
+/-  feather-icons
+:-  [%diary %$ %htmx]
+|=  dia=diary
+|=  =bowl:neo
+^-  manx
+|^
+  ;div.p2
+    =label  "Diary"
+    ;div.ma.fc.g2
+      =style  "max-width: 650px;"
+      ;+  form-put-entry
+      ;*
+      %+  turn
+        ~
+      link-entry
+    ==
+  ==
+++  form-put-entry
+  ;form.fc.g2
+    =style         "margin-bottom: 30px;"
+    =hx-post       "{(en-tape:pith:neo :(weld /neo/hawk here.bowl))}?stud=diary-diff"
+    =hx-on-submit  "this.reset()"
+    =hx-target     "this"
+    =hx-swap       "afterend"
+    =head          "put-entry"
+    ;date-now(name "id");
+    ;textarea.p2.bd1.br1
+      =name  "text"
+      =placeholder  "today, i ..."
+      =oninput  "this.setAttribute('value', this.value)"
+      =rows  "5"
+      =required  ""
+      =autocomplete  "off"
+      ;
+    ==
+    ;button.p2.b1.br1.bd1.wfc.hover.loader
+      ;span.loaded.s2: create
+      ;span.loading
+        ;+  loading.feather-icons
+      ==
+    ==
+  ==
+++  link-entry
+  |=  [pax=pith =pail:neo]
+  =/  tape  (trip !<(@t q.pail))
+  =/  subject-end  (fall (find [10]~ tape) 56)
+  =/  subject  (scag subject-end tape)
+  =/  id  (trip (snag 0 (pout pax)))
+  ;div.fr.g2
+    ;a.p2.br1.grow.b1.hover.loader
+      =href  "{(en-tape:pith:neo (weld /neo/hawk here.bowl))}/{id}"
+      ;div.loaded.fc.g1.js.as.g2
+        ;span.f3: {(pretty-date `@da`->:pax)}
+        ;span.bold: {subject}
+      ==
+      ;span.loading
+        ;+  loading.feather-icons
+      ==
+    ==
+    ;button.p2.br1.fr.g2.b1.hover.fc.ac.jc.loader
+      =onclick  "alert('not yet implemented. no tombstoning?')"
+      ;span.loaded
+        ;+  close.feather-icons
+      ==
+      ;span.loading
+        ;+  loading.feather-icons
+      ==
+    ==
+  ==
+++  pretty-date
+  |=  date=@da
+  ^-  tape
+  =/  d  (yore date)
+  "{(y-co:co y:d)}-{(y-co:co m:d)}-{(y-co:co d:t:d)}"
+--
+```
+
+There’s only a bit more going on here than in Counter’s `/con/number-htmx.hoon` file. The main thing is that we have `+link-entry` and `+pretty-date` helper arms which are being used like functional frontend components.
+
+The Diary frontend is a text box at the top with a list of entries generated from a list of the `/diary` shrub’s child `pith`s. Nowhere is a list of children in the state being passed from the backend to the frontend, like you might see with a Gall agent passing converting a `(list item)` to a JSON array through a mark in the event that a frontend has scried or subscribed on a path, all of which would require painstaking specification by the developer.
```

Comment:
While writing these tutorials there was this entropic process where I started using the shorthand `/diary` to refer to `/imp/diary.hoon` similar to how people use `%docket` to refer to `/app/docket.hoon`. I quite like it, it's path maximalism, but I was kind of just trying stuff and I'm not strongly attached to it at all; should commit to using it everywhere or nowhere.